### ISSUE TITLE
AutoSKLearn rank metric should learn a parameter to scale the distances

### ISSUE THREAD

**Issue by bjschoenfeld** [byu-dml/d3m-dynamic-neural-architecture#83]

**Comment by bjschoenfeld** [comment]
sklearn KNN handles 0 distances specially by using the discrete metric. If the distance is 0, the weight is 1; otherwise the weight is 0.

https://github.com/scikit-learn/scikit-learn/blob/7813f7efb5b2012412888b69e73d76f2df2b50b6/sklearn/neighbors/base.py#L78-L97

**Comment by bjschoenfeld** [comment]
I think the knn we are using is unsupervised. What if for the rank problem, we use supervised knn regression with pipeline score as the target? That is pretty standard. We could then also do the regression task.
